tests/itests/src/test/java/org/codice/ddf/admin/query/request/GraphQlHelper.java

```diff
@@ -0,0 +1,167 @@
+                                .get(graphQlSchemaEndpoint)
+                                .then()
+                                .extract()
+                                .jsonPath()
+                                .get("data") != null;
+
+                    } catch (Exception e) {
+                        return false;
+                    }
+                });
+    }
+
+    private String getResourceAsString(String filePath) {
+        try {
+            return IOUtils.toString(resourceClass.getResourceAsStream(filePath), "UTF-8");
+        } catch (IOException e) {
+            fail("Unable to retrieve resource: " + filePath);
+        }
+
+        return null;
+    }
+
+    public class GraphQLRequest {
+        private String queryResourcePath;
+
+        private String mutationResourcePath;
+
+        private String mutationFile;
+
+        private String queryFile;
+
+        private String graphQlEndpoint;
+
+        private Map<String, Object> reqArgs;
+
+        private ExtractableResponse response;
+
+        public GraphQLRequest(String queryResourcePath, String mutationResourcePath,
+                String graphQlEndpoint) {
+            this.queryResourcePath = queryResourcePath;
+            this.mutationResourcePath = mutationResourcePath;
+            this.graphQlEndpoint = graphQlEndpoint;
+            this.reqArgs = new HashMap<>();
+        }
+
+        public GraphQLRequest addArgument(String argName, Object argValue) {
+            reqArgs.put(argName, argValue);
+            return this;
+        }
+
+        public GraphQLRequest usingQuery(String queryFileName) {
+            this.queryFile = queryFileName;
+            return this;
+        }
+
+        public GraphQLRequest usingMutation(String mutationFileName) {
+            this.mutationFile = mutationFileName;
+            return this;
+        }
+
+        public GraphQLRequest send() {
+            Map<String, String> query = new HashMap<>();
+
+            String queryBody = null;
+
+            if (queryFile != null) {
+                queryBody = getResourceAsString(queryResourcePath + queryFile);
+            } else if (mutationFile != null) {
+                queryBody = getResourceAsString(mutationResourcePath + mutationFile);
+            } else {
+                fail("Failed to send GraphQLRequest. A query or mutation file must be specified before attempting to send the request.");
+            }
+            query.put("query", queryBody);
+
+            if (!reqArgs.isEmpty()) {
+                query.put("variables", Boon.toJson(reqArgs));
+            }
+
+            String queryStr = Boon.toPrettyJson(query);
+            LOGGER.debug("\nSending request:\n{}", queryStr);
+
+            response = given().when()
+                    .auth()
+                    .basic("admin", "admin")
+                    .body(queryStr)
+                    .post(graphQlEndpoint)
+                    .then()
```

Comment:
A lot of tests are using this method as a wait. If I use the `.statusCode` method here, it was do an assert and fail the test. We could very easily retrieve the status code and add another variable to the GraphQlRequest like we do the with the response. I'd rather have someone else who needs it do it though.